src/build_context.rs

```diff
@@ -137,14 +108,14 @@ impl BuildContext<CombinedValidator> {
     /// at the same time.
     pub fn into_slots_val(self) -> PyResult<Vec<CombinedValidator>> {
         let self_clone = self.clone();
-        self.slots
+        self.definitions
             .into_iter()
-            .map(|slot| match slot.op_val_ser {
-                Some(mut validator) => {
-                    validator.complete(&self_clone)?;
-                    Ok(validator)
-                }
+            .map(|slot| match slot {
                 None => py_err!("Slots Error: slot not yet filled"),
+                Some(mut v) => {
```

Comment:
can you move this above, and keep the name, it looks like the logic hasn't changed.